### ISSUE TITLE
Is recent FreeMiNT kernel slower than Helmut / 1.12 version?

### ISSUE THREAD

**Issue by mikrosk** [freemint/freemint#254]
As @Landemarre said on Discord:

_We have discuss regression of XaAES recently, me my concern is Mint kernel, not really regression but a lost of performance under very low power computer (8Mhz), it not concern a lot of people I think but it is fun to try on such configuration, when I do bench test on my own AES, I notice high difference of speed between Mint from Helmut branch and Mint official branch. I tested it under Hatari and Helmut branch give far better result than current branch (I think old Mint 1.12 give better results too), and this looks like relative to switch process or more process running. As I can see I not found difference for switch  task in source code. Under Helmut branch, in U:\proc I not see process 1 (Mint) and process 3 (AES), I have process 2 (update), 4 for desktop and 5 for zcontrol. Under Mint official branch process 1 (Mint) and 3 (AES) appear, process 1 have same size as 2 (update). Could it be possible 2 thread of Mint could run or something like this that could explain lost of speed (in this case more than 20%) ?_

**Comment by GokMasE** [comment]
It does sound like an intriguing phenomenon when a performance drop of >20% is suggested. 
As for the difference in processes listed in U:\proc, that sounds rather peculiar too.

An interesting experiment might be to do some benchmarking on different releases/snapshots to try and diagnose if there is some specific place (or places) in time where 8mhz machines starts to take a noticeable performance hit.

Was there any mention of how the 20% drop was measured?

**Comment by mikrosk** [comment]
(Olivier had trouble sending this reply)

_Hello_

_I have report this but the case is specific, in fact I have notice this with MyAES but it is not the case with XaAES even if it is quite difficult to compare between Helmut version and 1.19 version because it is not possible to run XaAES under Helmut version, so I was doing comparison between Helmut Mint version and 1.19 version with MyAES and in this case difference is huge, NAES is near not impacted too._

_I have done some search and difference looks highly link to Fselect function called by MyAES, I have done a speed up improvement that fix at around 90% the difference, but really I not understand why it has so many effect on result._

_Test is based on frame rate to diplay an opengl redraw in Kronos, there is no system call, this is a good way to check impact of system on calculation speed and in this case the impact of AES. Results are very stable on 8Mhz system._

_So in conclusion I can't really said Mint 1.19 impact of 20% the speed of the computer that's not true, it impact it if using MyAES only._

_I have a patch for this._

_Olivier_

**Comment by mikrosk** [comment]
> I have a patch for this.

Shamefully I have to admit that I have Olivier's patch for testing since May and I haven't done anything with it yet. :-/ So if there is a volunteer for testing, ideally with different AESes, I can provide a build.

**Comment by mikrosk** [comment]
@Landemarre yep, all works here. :)

**Comment by Landemarre** [comment]
Hello

I have report this but the case is specific, in fact I have notice this 
with MyAES but it is not the case with XaAES even if it is quite 
difficult to compare between Helmut version and 1.19 version because it 
is not possible to run XaAES under Helmut version, so I was doing 
comparison between Helmut Mint version and 1.19 version with MyAES and 
in this case difference is huge, NAES is near not impacted too.
I have done some search and difference looks highly link to Fselect 
function called by MyAES, I have done a speed up improvement that fix at 
around 90% the difference, but really I not understand why it has so 
many effect on result.


Test is based on frame rate to diplay an opengl redraw in Kronos, there 
is no system call, this is a good way to check impact of system on 
calculation speed and in this case the impact of AES. Results are very 
stable on 8Mhz system.

So in conclusion I can't really said Mint 1.19 impact of 20% the speed 
of the computer that's not true, it impact it if using MyAES only.
I have a patch for this.

Olivier

**Comment by Landemarre** [comment]
Hello

I have done some bench tests, globally under MyAES at 8Mhz speed is 
reduced (except for VDI tests) of 10%, under XaAES and NAES speed reduce 
is very small at around 2% except for NAES event manager bench, reduce 
in some case can be up to 80%, while under MyAES it is as global 
reduction of 10%. For MyAES as a thread is running possible there is 
something different in Fselect or Syield (I think more in Fselect) 
between older version and new, I use a value of 40ms for background 
task. For NAES really strange.

Olivier
